### ISSUE TITLE
Deployment of static files does not work with --dry-run Jetty-12

### ISSUE THREAD

**Issue by lachlan-roberts** [eclipse/jetty.project#10055]
**Jetty version(s)**
12.0.x

**Description**
The output of dry run does not work when pasted into command line when using static content with `core-deploy` module.

**How to reproduce?**

``` bash
java -jar $JETTY_HOME/start.jar --add-to-start=core-deploy,http
mkdir webapps/ROOT
echo "hello world" > webapps/ROOT/index.html
java -jar $JETTY_HOME/start.jar --dry-run
# paste the output of the dry run
```

Output of --dry-run:
> /opt/java/jdk-17.0.6+10/bin/java -Djava.io.tmpdir=/tmp -Djetty.home=/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home -Djetty.base=/tmp/jetty-test --class-path /tmp/jetty-test/resources:/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/logging/slf4j-api-2.0.7.jar:/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/logging/jetty-slf4j-impl-12.0.0-SNAPSHOT.jar:/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/jetty-http-12.0.0-SNAPSHOT.jar:/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/jetty-server-12.0.0-SNAPSHOT.jar:/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/jetty-xml-12.0.0-SNAPSHOT.jar:/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/jetty-util-12.0.0-SNAPSHOT.jar:/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/jetty-io-12.0.0-SNAPSHOT.jar:/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/jetty-deploy-12.0.0-SNAPSHOT.jar org.eclipse.jetty.xml.XmlConfiguration 'java.version'='17.0.6' 'java.version.major'='17' 'java.version.micro'='6' 'java.version.minor'='0' 'java.version.platform'='17' 'jetty.base'='/tmp/jetty-test' 'jetty.base.uri'='file:///tmp/jetty-test' 'jetty.home'='/home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home' 'jetty.home.uri'='file:///home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home' 'jetty.webapp.addServerClasses'='org.eclipse.jetty.logging.,file:///home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/lib/logging/,org.slf4j.' 'runtime.feature.alpn'='true' 'slf4j.version'='2.0.7' /home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/etc/jetty-bytebufferpool.xml /home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/etc/jetty-threadpool.xml /home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/etc/jetty.xml /home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/etc/jetty-deploy.xml /home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/etc/jetty-http.xml --env core contextHandlerClass=org.eclipse.jetty.server.handler.ResourceHandler$ResourceContext /home/lachlan/webtide/jetty-12.0.x/jetty-home/target/jetty-home/etc/jetty-core-deploy.xml

```
2023-07-03 10:32:26.866:INFO :oejs.Server:main: jetty-12.0.0-SNAPSHOT; built: 2023-07-03T00:21:06.345Z; git: ec2dbe73a8fb29243820ad822074808c422d056c; jvm 17.0.6+10
2023-07-03 10:32:26.888:INFO :oejdp.ScanningAppProvider:main: Deployment monitor core in [file:///tmp/jetty-test/webapps/] at intervals 0s
2023-07-03 10:32:26.895:INFO :oejd.DeploymentManager:main: addApp: App@654f0d9c[core,null,/tmp/jetty-test/webapps/ROOT]
2023-07-03 10:32:26.898:WARN :oejd.DeploymentManager:main: Unable to reach node goal: started
java.lang.NullPointerException: No Handler created for App: App@654f0d9c[core,null,/tmp/jetty-test/webapps/ROOT]
        at org.eclipse.jetty.deploy.bindings.StandardDeployer.processBinding(StandardDeployer.java:35)
        at org.eclipse.jetty.deploy.AppLifeCycle.runBindings(AppLifeCycle.java:183)
        at org.eclipse.jetty.deploy.DeploymentManager.requestAppGoal(DeploymentManager.java:525)
        at org.eclipse.jetty.deploy.DeploymentManager.addApp(DeploymentManager.java:165)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider.pathAdded(ScanningAppProvider.java:289)
        at org.eclipse.jetty.deploy.providers.ContextProvider.pathAdded(ContextProvider.java:520)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider$1.pathAdded(ScanningAppProvider.java:67)
        at org.eclipse.jetty.util.Scanner.reportAddition(Scanner.java:842)
        at org.eclipse.jetty.util.Scanner.reportDifferences(Scanner.java:808)
        at org.eclipse.jetty.util.Scanner.scan(Scanner.java:715)
        at org.eclipse.jetty.util.Scanner.doStart(Scanner.java:603)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:171)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider.doStart(ScanningAppProvider.java:260)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.deploy.DeploymentManager.startAppProvider(DeploymentManager.java:595)
        at org.eclipse.jetty.deploy.DeploymentManager.doStart(DeploymentManager.java:266)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:171)
        at org.eclipse.jetty.server.Server.start(Server.java:601)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.server.Handler$Abstract.doStart(Handler.java:395)
        at org.eclipse.jetty.server.Server.doStart(Server.java:542)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:1983)
2023-07-03 10:32:26.901:WARN :oejx.XmlConfiguration:main: Unable to execute XmlConfiguration
java.lang.NullPointerException: No Handler created for App: App@654f0d9c[core,null,/tmp/jetty-test/webapps/ROOT]
        at org.eclipse.jetty.deploy.bindings.StandardDeployer.processBinding(StandardDeployer.java:35)
        at org.eclipse.jetty.deploy.AppLifeCycle.runBindings(AppLifeCycle.java:183)
        at org.eclipse.jetty.deploy.DeploymentManager.requestAppGoal(DeploymentManager.java:525)
        at org.eclipse.jetty.deploy.DeploymentManager.addApp(DeploymentManager.java:165)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider.pathAdded(ScanningAppProvider.java:289)
        at org.eclipse.jetty.deploy.providers.ContextProvider.pathAdded(ContextProvider.java:520)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider$1.pathAdded(ScanningAppProvider.java:67)
        at org.eclipse.jetty.util.Scanner.reportAddition(Scanner.java:842)
        at org.eclipse.jetty.util.Scanner.reportDifferences(Scanner.java:808)
        at org.eclipse.jetty.util.Scanner.scan(Scanner.java:715)
        at org.eclipse.jetty.util.Scanner.doStart(Scanner.java:603)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:171)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider.doStart(ScanningAppProvider.java:260)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.deploy.DeploymentManager.startAppProvider(DeploymentManager.java:595)
        at org.eclipse.jetty.deploy.DeploymentManager.doStart(DeploymentManager.java:266)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:171)
        at org.eclipse.jetty.server.Server.start(Server.java:601)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.server.Handler$Abstract.doStart(Handler.java:395)
        at org.eclipse.jetty.server.Server.doStart(Server.java:542)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:1983)
Exception in thread "main" java.lang.NullPointerException: No Handler created for App: App@654f0d9c[core,null,/tmp/jetty-test/webapps/ROOT]
        at org.eclipse.jetty.deploy.bindings.StandardDeployer.processBinding(StandardDeployer.java:35)
        at org.eclipse.jetty.deploy.AppLifeCycle.runBindings(AppLifeCycle.java:183)
        at org.eclipse.jetty.deploy.DeploymentManager.requestAppGoal(DeploymentManager.java:525)
        at org.eclipse.jetty.deploy.DeploymentManager.addApp(DeploymentManager.java:165)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider.pathAdded(ScanningAppProvider.java:289)
        at org.eclipse.jetty.deploy.providers.ContextProvider.pathAdded(ContextProvider.java:520)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider$1.pathAdded(ScanningAppProvider.java:67)
        at org.eclipse.jetty.util.Scanner.reportAddition(Scanner.java:842)
        at org.eclipse.jetty.util.Scanner.reportDifferences(Scanner.java:808)
        at org.eclipse.jetty.util.Scanner.scan(Scanner.java:715)
        at org.eclipse.jetty.util.Scanner.doStart(Scanner.java:603)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:171)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider.doStart(ScanningAppProvider.java:260)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.deploy.DeploymentManager.startAppProvider(DeploymentManager.java:595)
        at org.eclipse.jetty.deploy.DeploymentManager.doStart(DeploymentManager.java:266)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:171)
        at org.eclipse.jetty.server.Server.start(Server.java:601)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.server.Handler$Abstract.doStart(Handler.java:395)
        at org.eclipse.jetty.server.Server.doStart(Server.java:542)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:1983)

```

**Comment by lachlan-roberts** [comment]
I think the issue is that in `org.eclipse.jetty.server.handler.ResourceHandler$ResourceContext`, the `$ResourceContext` is interpreted as an environment variable.

I have tested by quoting this with single quotes and it works.

**Comment by gregw** [comment]
will be fixed by #10090

**Comment by sbordet** [comment]
Fixed by #10099, merged to 12.

After the merge, this is the output:

```
/home/simon/programs/openjdk-20.0.1+9/bin/java
-Djava.io.tmpdir=/tmp
-Djetty.home=/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home
-Djetty.base=/tmp/jetty.base
--class-path
/tmp/jetty.base/resources:/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/logging/slf4j-api-2.0.7.jar:/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/logging/jetty-slf4j-impl-12.0.0-SNAPSHOT.jar:/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/jetty-http-12.0.0-SNAPSHOT.jar:/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/jetty-server-12.0.0-SNAPSHOT.jar:/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/jetty-xml-12.0.0-SNAPSHOT.jar:/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/jetty-util-12.0.0-SNAPSHOT.jar:/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/jetty-io-12.0.0-SNAPSHOT.jar:/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/jetty-deploy-12.0.0-SNAPSHOT.jar
org.eclipse.jetty.xml.XmlConfiguration
java.version=20.0.1
jetty.base=/tmp/jetty.base
jetty.base.uri=file:///tmp/jetty.base
jetty.home=/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home
jetty.home.uri=file:///home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home
jetty.webapp.addServerClasses=org.eclipse.jetty.logging.,file:///home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/lib/logging/,org.slf4j.
runtime.feature.alpn=true
slf4j.version=2.0.7
/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/etc/jetty-bytebufferpool.xml
/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/etc/jetty-threadpool.xml
/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/etc/jetty.xml
/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/etc/jetty-deploy.xml
/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/etc/jetty-http.xml
--env
core
contextHandlerClass='org.eclipse.jetty.server.handler.ResourceHandler$ResourceContext'
/home/simon/opensource/jetty/12.0/jetty-home/target/jetty-home/etc/jetty-core-deploy.xml
```

In particular:

```
--env core contextHandlerClass='org.eclipse.jetty.server.handler.ResourceHandler$ResourceContext'
```

Shows that the `$` has been detected and the value properly quoted.
